src/sagas/bucket.js

```diff
@@ -133,35 +136,41 @@ export function* deleteCollection(getState, action) {
   }
 }
 
-export function* listBucketCollections(getState, action) {
+export function* listBucketCollections(getState: GetStateFn, action: Object): SagaGen {
   const {bid} = action;
   try {
     const bucket = getBucket(bid);
-    const {data} = yield call([bucket, bucket.listCollections]);
-    yield put(listBucketCollectionsSuccess(data));
+    const result = yield call([bucket, bucket.listCollections]);
+    if (result == null) {
+      throw new Error("Empty result.");
+    }
+    yield put(listBucketCollectionsSuccess(result.data));
   } catch(error) {
     yield put(notifyError("Couldn't list bucket collections.", error));
   }
 }
 
-export function* listBucketHistory(getState, action) {
+export function* listBucketHistory(getState: GetStateFn, action: Object): SagaGen {
   const {bid, filters: {resource_name, since}} = action;
   try {
     const bucket = getBucket(bid);
-    const {data} = yield call([bucket, bucket.listHistory], {
+    const result = yield call([bucket, bucket.listHistory], {
       since,
       filters: {
         resource_name,
         exclude_resource_name: "record"
       }
     });
-    yield put(listBucketHistorySuccess(data));
+    if (result == null) {
```

Comment:
It looks like a limitation/bug in flow... https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVUCuA7AxgFwEs5tl8BTAZ3wCYAKASgC4wBxc7cgJwEN84uAHmyYAtgCNuAGhAyAfGADeqMGACehcjAAmYAIwBuFes06wNIwF90uEtTCkAvGArV6DAHScAHvkZHUW2x7HhYRCW4wZ2wPADceGExyIyD7cRZYuEJdAB8HMUkuKIc4hKSDIA